modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/tree/BPlusTree.java

```diff
@@ -2447,16 +2457,20 @@ private T doPut(T row, boolean needOld) throws IgniteCheckedException {
                         continue;
 
                     case FOUND:
-                        // We may need to insert split key into upper level here.
+                        // We may need to perform an inner replace in the upper level.
                         if (!p.isFinished()) {
-                            // It must be impossible to have an insert higher than the current root,
-                            // because we are making decision about creating new root while keeping
-                            // write lock on current root, so it can't concurrently change.
-                            assert p.btmLvl <= getRootLevel();
+                            res = p.finishTail();
 
-                            checkInterrupted();
+                            // If not found, then the tree grew beyond our call stack -> retry from the actual root.
```

Comment:
what do you mean by `grew beyond our call stack`?

Reply:
This is a copy from already existing comment. Nonetheless, it means that tree height has increased while we were attempting this put operation.

Reply:
Can you update the comment, please?

Reply:
Done